libsolidity/ast/Types.h

```diff
@@ -1032,7 +1033,11 @@ class FunctionType: public Type
 	/// expression the function is called on.
 	bool canTakeArguments(TypePointers const& _arguments, TypePointer const& _selfType = TypePointer()) const;
 	/// @returns true if the types of parameters are equal (does't check return parameter types)
-	bool hasEqualArgumentTypes(FunctionType const& _other) const;
+	bool hasEqualParameterTypes(FunctionType const& _other) const;
+	/// @returns true iff the return types are equal (does't check parameter types)
+	bool hasEqualReturnTypes(FunctionType const& _other) const;
+	/// @returns true iff the parameter types and the return types are equal
+	bool hasEqualParameterAndReturnTypes(FunctionType const& _other) const;
```

Comment:
I think these could also be pulled out and even used by the "function shadowing" type checker feature.